<launch>

  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="false"/>
  <arg name="gui" default="false"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/> 
  <arg name="verbose" default="false"/>

  <!-- JY901B IMU参数设置 -->
  <!-- <arg name="imu_port" default="/dev/ttyUSB0" doc="JY901B串口设备路径，可能是/dev/ttyUSB0或/dev/ttyACM0"/>
  <arg name="imu_frame_id" default="imu_link" doc="IMU坐标系ID"/>
  <arg name="imu_rate" default="100" doc="IMU数据发布频率"/> -->

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <!--include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find robocon25_sim)/worlds/robocon25.world"/>
    <arg name="verbose" 	value="true"/>
    <arg name="paused"		value="false"/>
  </include-->
  
  <!-- Spawn a robot into Gazebo -->
  <!--node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-file $(find robocon25_sim)/models/robot.model -urdf -model robot" /-->

  <!-- [定位方法一] 只用场线定位 localization 节点-->
  <!--node name="localization" pkg="conqu_localization" type="localization" output="screen">
    <param name= "table_file" value="$(find conqu_localization)/config/dist_table.txt" />
    <param name= "lines_file" value="$(find conqu_localization)/config/field_lines.jpg" />
    <param name= "field_file" value="$(find conqu_localization)/config/field_bg.png" />
  </node-->

  <!-- [定位方法二] 使用边线辅助场线定位 loc_sidelines 场线定位节点 -->
  <node name="loc_sidelines" pkg="conqu_localization" type="loc_sidelines">
    <param name="blue_h_min" value="100"/>
    <param name="blue_h_max" value="145"/>
    <param name="red_h_min" value="127"/>
    <!-- <param name="red_h_max" value="3"/> -->
    <param name="gray_s_threhold" value="70"/>
    <param name="gray_v_threhold" value="90"/>
    <param name="white_v_min" value="150"/>
    <param name="white_v_max" value="180"/>

    <param name= "table_file" value="$(find conqu_localization)/config/dist_small" />
    <param name= "lines_file" value="$(find conqu_localization)/config/field_lines.jpg" />
    <param name= "field_file" value="$(find conqu_localization)/config/field_bg.png" />
    <param name= "lines_map_file" value="$(find conqu_localization)/config/white_lines.png"/>
    <param name= "red_map_file" value="$(find conqu_localization)/config/red_lines.png"/>
    <param name= "blue_map_file" value="$(find conqu_localization)/config/blue_lines.png"/>
  </node>

  <!-- 启动 relocalization 重定位节点 -->
  <!-- <node name="relocalization" pkg="conqu_localization" type="relocalization">
    <param name= "table_file" value="$(find conqu_localization)/config/dist_small" />
    <param name= "lines_file" value="$(find conqu_localization)/config/field_lines.jpg" />
    <param name= "field_file" value="$(find conqu_localization)/config/field_bg.png" />
    <param name= "lines_map_file" value="$(find conqu_localization)/config/white_lines.png"/>
    <param name= "red_map_file" value="$(find conqu_localization)/config/red_lines.png"/>
    <param name= "blue_map_file" value="$(find conqu_localization)/config/blue_lines.png"/>
  </node> -->

  <!-- 调用usb摄像头 -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node">
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="camera_link" />
    <param name="autowhitebalance" value="true" />
    <param name="io_method" value="mmap"/>
    <!-- <remap from="/usb_cam/image_raw" to="/omni_camera/image_raw"/> -->
  </node>

  <!-- 启动JY901B IMU节点 -->
  <!-- <node name="jy901_node" pkg="conqu_localization" type="jy901_node" output="screen">
    <param name="port" value="$(arg imu_port)"/>
    <param name="baud" value="9600"/>
    <param name="frame_id" value="$(arg imu_frame_id)"/>
    <param name="frequency" value="$(arg imu_rate)"/>
  </node> -->

  <!-- 添加IMU姿态估计滤波器 -->
  <!-- <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_node" output="screen">
    <param name="use_mag" value="true"/>
    <param name="publish_tf" value="false"/>
    <param name="world_frame" value="enu"/>
    <param name="orientation_stddev" value="0.01"/>
    <remap from="/imu/data_raw" to="/imu/data"/>
    <remap from="/imu/data" to="/imu/data_filtered"/>
  </node> -->

</launch>